platform/mbed_error.c

```diff
@@ -556,6 +590,51 @@ static void print_error_report(const mbed_error_ctx *ctx, const char *error_msg,
     mbed_stats_sys_get(&sys_stats);
     mbed_error_printf("\nFor more info, visit: https://mbed.com/s/error?error=0x%08X&osver=%" PRId32 "&core=0x%08" PRIX32 "&comp=%d&ver=%" PRIu32 "&tgt=" GET_TARGET_NAME(TARGET_NAME), ctx->error_status, sys_stats.os_version, sys_stats.cpu_id, sys_stats.compiler_id, sys_stats.compiler_version);
 #endif
+
+#if MBED_STACK_DUMP_ENABLED && defined(MBED_CONF_RTOS_PRESENT)
+#define STACK_END_MARK		0xCCCCCCCC
+#define STACK_END_MARK_CNT	3
```

Comment:
what does this number do (why is it 3) ? line 613 is not that clear to me - might be worth adding a comment there what we do there ?

Reply:
Added comment (https://github.com/ARMmbed/mbed-os/pull/10872/files#diff-ae57d671f3af44d01b722ed277444e27R561) to elaborate the stack end detection logic.